app/src/test/java/com/github/swent/echo/compose/authentication/CreateProfileTest.kt

```diff
@@ -62,4 +83,80 @@ class CreateProfileTest {
         composeTestRule.onNodeWithTag("BA1").assertExists()
         composeTestRule.onNodeWithTag("BA2").assertExists()
     }
+
+    @Test
+    fun profileCreationPictureIsCorrect() {
+        val picture = Bitmap.createBitmap(1000, 1000, Bitmap.Config.ARGB_8888)
+        var changedPicture: Bitmap? = picture
+        composeTestRule.setContent {
+            ProfilePictureEdit(picture = picture, onPictureChange = { changedPicture = it })
+        }
+        composeTestRule
+            .onNodeWithTag("profile-picture-image")
+            .assertIsDisplayed()
+            .assertHasClickAction()
+        composeTestRule
+            .onNodeWithTag("profile-picture-delete")
+            .assertIsDisplayed()
+            .assertHasClickAction()
+        composeTestRule.onNodeWithTag("profile-picture-delete").performClick()
+        assertNull(changedPicture)
+        composeTestRule.onNodeWithTag("profile-picture-image").performClick()
+    }
+
+    @Test
+    fun profileCreationPictureTransformerCallbacksTest() {
+        val picture = Bitmap.createBitmap(300, 1000, Bitmap.Config.ARGB_8888)
+        var changedPicture: Bitmap = picture
+        var canceled = false
+        composeTestRule.setContent {
+            PictureTransformer(
+                picture = picture,
+                onConfirm = { changedPicture = it },
+                onCancel = { canceled = true }
+            )
+        }
+        composeTestRule.onNodeWithTag("profile-picture-transformer").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("profile-picture-transformer-confirm").performClick()
+        assertTrue(changedPicture != picture)
+        assertTrue(changedPicture.height == changedPicture.width)
+        composeTestRule.onNodeWithTag("profile-picture-transformer-cancel").performClick()
+        assertTrue(canceled)
+    }
+
+    @Test
+    fun profileCreationPictureTransformerGestureTest() {
+        val picture = Bitmap.createBitmap(300, 1000, Bitmap.Config.ARGB_8888)
+        var changedPicture: Bitmap = picture
+        composeTestRule.setContent {
+            PictureTransformer(
+                picture = picture,
+                onConfirm = { changedPicture = it },
+                onCancel = {}
+            )
+        }
+        val image = composeTestRule.onNodeWithTag("profile-picture-image")
+        image.assertIsDisplayed()
+        image.performTouchInput { this.swipe(Offset.Zero, Offset(50f, 50f)) }
+        image.assertPositionInRootIsEqualTo(15.dp, 15.dp)
+    }
+
+    // this test needs to be run there because of the Bitmap class which require android
```

Comment:
Knitpick: the comment is a bit confusing: what exactly do you mean by 'test needs to be run there'? run where? Maybe I am just confused for nothing here...

Reply:
ok I will remove this comment